### PR TITLE
refactor: Plan models

### PR DESCRIPTION
**Pull request by chrisgacsal**
## Overview

Refactor Plan API by moving common types/models under the `productcatalog` in order to make them reusable in Subscription and Invoicing APIs.

### CODE REVIEW

**Review by GAlexIHU — APPROVED** [review]
👍

File: openmeter/productcatalog/phase.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	Description *string `json:"description,omitempty"`
+
+	// Metadata stores user defined metadata for Phase.
+	Metadata Metadata `json:"metadata,omitempty"`
+
+	// StartAfter defines the time period after the Phase becomes active measured from the Subscription start.
+	StartAfter datex.Period `json:"interval"`
+
+	// Discounts stores a set of discount(s) applied to all or specific RateCards.
+	Discounts Discounts `json:"discounts,omitempty"`
```
[the remark this answers is not in the document]
Comment (tothandras) [reply openmeter/productcatalog/phase.go:36]: I think this one is correct: `type Discounts []Discount`

[the remark this answers is not in the document]
Comment (chrisgacsal) [reply openmeter/productcatalog/phase.go:36]: @GAlexIHU If you mean that the `Discounts` should be part of the `Phase` instad of the `PhaseMeta` the smae way as the `RateCards` then I agree.

File: openmeter/productcatalog/metadata.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"maps"
+
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+var _ models.Equaler[Metadata] = (*Metadata)(nil)
+
+type Metadata map[string]string
```
[the remark this answers is not in the document]
Comment (chrisgacsal) [reply openmeter/productcatalog/metadata.go]: Agreed. Moved it to models.

File: openmeter/productcatalog/ratecard.go [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+	Key string `json:"key"`
+
+	// Name of the RateCard
+	Name string `json:"name"`
+
+	// Description for the RateCard
+	Description *string `json:"description,omitempty"`
+
+	// Metadata a set of key/value pairs describing metadata for the RateCard
+	Metadata Metadata `json:"metadata,omitempty"`
```
[the remark this answers is not in the document]
Comment (chrisgacsal) [reply openmeter/productcatalog/ratecard.go]: Yeah, so we have metadata on Plan/Phase/RateCard level. I do not think it is an issue to allow users to provide extra info on these types. So I'd just keep it as is for now.

[the remark this answers is not in the document]
Comment (GAlexIHU) [reply openmeter/productcatalog/ratecard.go]: Sry, i meant the `plan` package, my point was that i don't see this metadata being shared accross `plan`, `subs`, `billing`, they would all be separate metadata, or wouldn't they?

File: openmeter/productcatalog/phase.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisgacsal) [reply openmeter/productcatalog/phase.go]: Done.

**Review by tothandras — APPROVED** [review]

**Review by GAlexIHU — APPROVED** [review]

**Pull request merged**